tests/grpc/mock_v1beta1/kmsc_mock_test.go

```diff
@@ -0,0 +1,115 @@
+package mock_v1beta1_test
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"testing"
+
+	kmscmock "github.com/Azure/kubernetes-kms/tests/grpc/mock_v1beta1"
+	k8spb "github.com/Azure/kubernetes-kms/v1beta1"
+	"github.com/golang/mock/gomock"
+	"github.com/golang/protobuf/proto"
+	"golang.org/x/net/context"
+)
+
+var (
+	// TODO get the version automatically from the folder name
+	version = "v1beta1"
+)
+
+// func setup(t *testing.T) (*gomock.Controller, kmscmock.MockKeyManagementServiceClient) {
+// 	ctrl := gomock.NewController(t)
+// 	return nil, kmscmock.NewMockKeyManagementServiceClient(&ctrl)
+
+// }
+
+// rpcMsg implements the gomock.Matcher interface
+type rpcMsg struct {
+	msg proto.Message
+}
+
+func (r *rpcMsg) Matches(msg interface{}) bool {
+	m, ok := msg.(proto.Message)
+	if !ok {
+		return false
+	}
+	return proto.Equal(m, r.msg)
+}
+
+func (r *rpcMsg) String() string {
+	return fmt.Sprintf("is %s", r.msg)
+}
+
+//Version
+func TestVersion(t *testing.T) {
+	//ctrl, mockKeyManagementServiceClient := setup(t)
+	ctrl := gomock.NewController(t)
+	mockKeyManagementServiceClient := kmscmock.NewMockKeyManagementServiceClient(ctrl)
+	defer ctrl.Finish()
+	req := &k8spb.VersionRequest{Version: version}
+	mockKeyManagementServiceClient.EXPECT().Version(
+		gomock.Any(),
+		&rpcMsg{msg: req},
+	).Return(&k8spb.VersionResponse{Version: version}, nil)
+	ask := "v1beta1"
+	r, _ := mockKeyManagementServiceClient.Version(context.Background(), &k8spb.VersionRequest{Version: ask})
+	if r != nil {
+		t.Logf("Test passed, ask : %s answer : %s", version, ask)
+	}
+}
+
+func TestBadVersion(t *testing.T) {
```

Comment:
I am not sure if I am doing the right thing here since I am mocking a good and a Bad request (below) with expecting return...

Reply:
This mock is for the gRPC contract. Let's keep investigating how to mock key vault to test that part of the code.